packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -119,11 +119,14 @@ export async function fetchSnapshotWithRedeem(
 	snapshotDownloader: (
 		finalOdspResolvedUrl: IOdspResolvedUrl,
 		storageToken: string,
+		groupIds: string[],
 		snapshotOptions: ISnapshotOptions | undefined,
 		controller?: AbortController,
 	) => Promise<ISnapshotRequestAndResponseOptions>,
 	putInCache: (valueWithEpoch: IVersionedValueWithEpoch) => Promise<void>,
 	removeEntries: () => Promise<void>,
+	groupIds: string[],
+	fetchSnapshotForInitialLoad: boolean,
```

Comment:
Isn't fetchSnapshotForInitialLoad === (groupIds.length === 0)? I.e. we have two ways to encode same thing, which could lead to bugs (i.e. someone calling with fetchSnapshotForInitialLoad  = true and non-empty groupIds)

Reply:
This is used by both old and new flows. Old flow passes this as true when latest snapshot needs to be fetched and groupids as empty while the newer flow passes this as true, in case it is setting group id "default" and length is 1 in that case.

Reply:
Hm, maybe my comment is not entirely true, as defaultGroupIdForSnapshot could be used (not sure if it applies at this layer or not). Please read other comments (I learn as I go through code)